pkg/pipeline/prepare_pipelines.go

```diff
@@ -225,22 +228,76 @@ func (ctx *pipelineContext) pipelineFileExists(pipelineFilePath string) error {
 	return err
 }
 
-func (ctx *pipelineContext) buildTasks(envName string, tasks []pipelinev1.Task, timestamp string) map[string]pipelinev1.Task {
+func (ctx *pipelineContext) buildTasks(envName string, tasks []pipelinev1.Task, timestamp string) (map[string]pipelinev1.Task, error) {
+	var errs []error
 	taskMap := make(map[string]pipelinev1.Task)
 	for _, task := range tasks {
 		originalTaskName := task.Name
 		taskName := fmt.Sprintf("radix-task-%s-%s-%s-%s", getShortName(envName), getShortName(originalTaskName), timestamp, ctx.hash)
-		task.ObjectMeta.Name = taskName
-		task.ObjectMeta.Annotations = map[string]string{defaults.PipelineTaskNameAnnotation: originalTaskName}
-		task.ObjectMeta.Labels = labels.GetLabelsForEnvironment(ctx, envName)
+		if task.ObjectMeta.Labels == nil {
+			task.ObjectMeta.Labels = map[string]string{}
+		}
+		if task.ObjectMeta.Annotations == nil {
+			task.ObjectMeta.Annotations = map[string]string{}
+		}
+
+		for k, v := range labels.GetLabelsForEnvironment(ctx, envName) {
+			task.ObjectMeta.Labels[k] = v
+		}
+
+		if val, ok := task.ObjectMeta.Labels[labels.AzureWorkloadIdenityUse]; ok && val == "true" {
+			validateAzureSkipContainers(&task)
+		}
+
+		task.ObjectMeta.Annotations[defaults.PipelineTaskNameAnnotation] = originalTaskName
 		if ctx.ownerReference != nil {
 			task.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*ctx.ownerReference}
 		}
 		ensureCorrectSecureContext(&task)
+
+		task.ObjectMeta.Name = taskName
 		taskMap[originalTaskName] = task
 		log.Debugf("created the task %s", task.Name)
 	}
-	return taskMap
+	return taskMap, errors.Join(errs...)
+}
+
+func validateAzureSkipContainers(task *pipelinev1.Task) {
```

Comment:
```suggestion
func sanitizeAzureSkipContainersAnnotation(task *pipelinev1.Task) {
```
Is this a more descriptive name?